plugins/volume/org.xen.xapi.storage.fsp/volume.py

```diff
@@ -11,62 +11,51 @@
 from xapi.storage.libs import util
 from xapi.storage.libs.libcow.callbacks import VolumeContext
 from xapi.storage.libs.libcow.imageformat import ImageFormat
-from xapi.storage.libs.libcow.lock import PollLock
+from xapi.storage.libs.libcow.lock import Lock, PollLock
+from xapi.storage.libs.libcow.volume import _vdi_sanitize
 from xapi.storage.libs.libcow.volume_implementation import Implementation as \
     DefaultImplementation
 
 
 @util.decorate_all_routines(util.log_exceptions_in_function)
 class Implementation(DefaultImplementation):
-    def create(self, dbg, sr, name, description, size, sharable):
-        # WORKAROUND: For the moment we can't use a config param to forward the
-        # shared_dir string. We use the name field instead.
-        # We must open few PRs in the upstream repositories to support that:
-        # - xenopsd
-        # - xapi-storage-script
-        #
-        # TODO: The VDI size is useless. So it would be more interesting to add
-        # a VDI.import method instead of using VDI.create.
+    def destroy(self, dbg, sr, key):
+        meta = util.get_sr_metadata(dbg, 'file://' + sr)
+        cb = self.callbacks
+        with VolumeContext(cb, sr, 'w') as opq:
+            with Lock(opq, 'gl', cb):
+                with cb.db_context(opq) as db:
+                    vdi = db.get_vdi_by_id(key)
+                    db.delete_vdi(key)
+                with cb.db_context(opq) as db:
+                    cb.volumeDestroy(opq, str(vdi.volume.id))
+                    db.delete_volume(vdi.volume.id)
```

Comment:
You can remove this duplication calling the parent method with `super`.
And probably better: you can delete this full `destroy` impl, because I don't see specific new action here. :sweat_smile: